app/assets/javascripts/components/CartedSubscriptions.jsx

```diff
@@ -0,0 +1,111 @@
+var CartedSubscriptions = React.createClass({
+  getInitialState: function(){
+     return {
+       initialItems: this.props.cartedSubscriptions,
+       carted_subscriptions: [],
+       initialSubscriptionsTotal: this.props.subscriptionsTotal,
+       isEmpty: false
+     }
+  },
+  componentWillMount: function(){
+    this.setState({carted_subscriptions: this.state.initialItems, subscriptions_total: this.state.initialSubscriptionsTotal, isEmpty: this.state.isEmpty})
+  },
+  calcTotal: function() {
+    console.log("calculating total" + this.state.subscriptions_total);
+    var total = 0;
+    this.state.carted_subscriptions.forEach(function(carted_subscription) {
+      console.log(carted_subscription);
+      total += (carted_subscription.price * carted_subscription.quantity);
+    });
+    console.log("FINISHED CALCULATING.." + total);
+    this.setState({subscriptions_total: total});
+  },
+  updateQuantity: function(val, id) {
+    console.log(val, "from parent && id: ", id);
+    var that = this;
+    $.ajax({
+      type: "PATCH",
+      url: "/api/carted_subscriptions/" + id + "/" + val,
+      contentType: "application/json; charset=utf-8",
+      dataType: "json",
+      success: function(result){
+        if (result.error) {
+          alert(result.error);
+        } else {
+          that.state.carted_subscriptions.map((el) =>
+            (el.id == id) ? (
+              el.quantity = parseInt(val)
+            ) : (
+              el
+            )
+          );
+          console.log("update success");
+          that.calcTotal();
+          that.setState({carted_subscriptions: that.state.carted_subscriptions});
+        }
+      }
+    });
+  },
+  deleteItem: function(id){
+    this.state.carted_subscriptions = this.state.carted_subscriptions.filter((carted_subscription) =>
+      carted_subscription.id !== id
+    );
+    var that = this;
+    $.ajax({
+      type: "DELETE",
+      url: "/api/carted_subscriptions/" + id,
+      contentType: "application/json; charset=utf-8",
+      dataType: "json",
+      success: function(result){
+        console.log(result);
+        that.isCartEmpty();
+      }
+    });
+    this.setState({carted_subscriptions: this.state.carted_subscriptions});
+  },
+  isCartEmpty: function() {
+    if (this.state.carted_subscriptions.length == 0) {
```

Comment:
super nitpick, but you should use `===`, some linters or code coverage tools will give you a warning.